handlers/k8sconfig_handler.go

```diff
@@ -293,41 +301,69 @@ func (h *Handler) GetCurrentContext(token string, prov models.Provider) (*models
 				logrus.Warn("failed to save the context: ", err)
 				continue
 			}
+			if !viper.GetBool("SKIP_COMP_GEN") {
+				k8ctxID := ctx.ID
+				cfg, err := ctx.GenerateKubeConfig()
+				if err != nil {
+					return &ctx, nil
+				}
 
-			if ctx.IsCurrentContext {
-				_, _ = prov.SetCurrentContext(token, ctx.ID) // Ignore the error
-
-				if !viper.GetBool("SKIP_COMP_GEN") {
-					ctxID := ctx.ID
-					cfg, err := ctx.GenerateKubeConfig()
+				go func(config []byte, ctx string) {
+					compCreationSingleton.cancelPreviousRun(ctx) //if a registerK8sComponents is still running for the same context then we safely cancel that assuming that run to be stale now
```

Comment:
I might not be reading this in the right "context" ( 🙂 ). Will you confirm that Meshery Server will not cancel an existing component generation process upon Meshery UI refresh, but instead, Meshery Server will finish it's current generation task? 

Will you confirm that Meshery Server will only generate Kubernetes components upon `create` of a K8s context?

Reply:
To second question- yes, components are generated on create of a k8s context.
On first question,  no component generation will not be cancelled due to UI refresh.
It will be cancelled when context is reloaded or deleted. Because for reloads or deletes of a k8s context, components generated from it will not be used anywhere anyways and in that case we are wasting computing resources.

Reply:
In case of reloads, the component generation will restart with the new reloaded k8s context.